tidy3d/plugins/smatrix/component_modelers/base.py

```diff
@@ -35,7 +36,7 @@ class AbstractComponentModeler(ABC, Tidy3dBaseModel):
         description="Simulation describing the device without any sources present.",
     )
 
-    ports: Tuple[Union[Port, LumpedPort], ...] = pd.Field(
+    ports: Tuple[Union[Port, AbstractTerminalPort], ...] = pd.Field(
```

Comment:
typically it's safer to annotate with a `Union[]` of all possible types, instead of the abstract type. So there would be like a `TerminalPortType = Union[...]` and then `Union[Port, TerminalPortType]`